File: app/src/main/java/de/tum/in/tumcampusapp/models/tumo/OrgDetailItemList.java
package de.tum.in.tumcampusapp.models.tumo;

import org.simpleframework.xml.ElementList;
import org.simpleframework.xml.Root;

import java.util.List;

/**
 * This class is dealing with the de-serialization of the output of TUMOnline to the method "sucheLehrveranstaltungen".
 */

@Root(name = "rowset")
public class OrgDetailItemList {

    /**
     * List of OrganisationItems
     */
    @ElementList(inline = true, required = false)
    private List<OrgDetailsItem> groups;

    public List<OrgDetailsItem> getGroups() {
        return groups;
    }
}

File: app/src/main/java/de/tum/in/tumcampusapp/component/other/departments/model/OrgDetailItemList.kt
package de.tum.`in`.tumcampusapp.component.other.departments.model

import com.tickaroo.tikxml.annotation.Element
import com.tickaroo.tikxml.annotation.Xml

/**
 * This class is dealing with the de-serialization of the output of TUMOnline to the method "sucheLehrveranstaltungen".
 */

@Xml(name = "rowset")
data class OrgDetailItemList(@Element var groups: List<OrgDetailsItem> = mutableListOf())
